### PR TITLE
MAINT, TYP: Added missing where typehints in fromnumeric.pyi

### PR DESCRIPTION
**Pull request by janusheide**
Adding typehints for 'where' argument in a couple of functions in fromnumeric, where it appears to be missing.

### CODE REVIEW

File: numpy/core/fromnumeric.pyi [hunk 1]
```diff
@@ -248,13 +248,15 @@ def all(
     axis: None = ...,
     out: None = ...,
     keepdims: Literal[False] = ...,
+    where: _ArrayLikeBool_co = ...,
```
Comment (BvB93) [inline numpy/core/fromnumeric.pyi:251]:
```suggestion
    *,
    where: _ArrayLikeBool_co = ...,
```
As `where` consistently seems to be keyword-only. Note that the same comment applies to all functions here.

**Review by BvB93 — COMMENTED** [review]
Thanks, looks pretty good overall but I've got two general comments:
* The `where` arguments should be marked as keyword only (matching runtime behavior).
* Besides the functions in `fromnumeric.pyi` there are also the identically named methods in the main [`__init__.pyi`](https://github.com/numpy/numpy/blob/main/numpy/__init__.pyi) file (_e.g._ `ndarray.all`). Could you change those as well?

**Comment by janusheide** [comment]
@BvB93 If I understood correctly, your comments should have been addressed now.

I note that there are a bunch of functions in fromnumeric.py, where the *where* argument is not keyword-only, e.g. https://github.com/numpy/numpy/blob/8f8e14ef6150100bf2872a449877f97897b18b06/numpy/core/fromnumeric.py#L2163, I assume this inconsistency is not intentional - and maybe I can use it as an excuse for missing it in the other functions ;)

Wrt. the kw only inconsistency, would you like me to open an issue, a PR or both, or do nothing?

**Comment by BvB93** [comment]
> Wrt. the kw only inconsistency, would you like me to open an issue, a PR or both, or do nothing?

So the move towards the use of keyword-only arguments is somewhat recent, and with this in mind it is no coincidence that the (keyword-only) `where` arguments addressed in this PR were only added back in numpy 1.20.

Changing all the (much older) remaining `where` arguments to keyword-only would be technically be backwards compatible. While I doubt they're used much as a positional parameter out in the wild, it's probably not the risk of breaking things.

**Comment by BvB93** [comment]
Thanks @janusheid

**Comment by janusheide** [comment]
> > Wrt. the kw only inconsistency, would you like me to open an issue, a PR or both, or do nothing?
> 
> So the move towards the use of keyword-only arguments is somewhat recent, and with this in mind it is no coincidence that the (keyword-only) `where` arguments addressed in this PR were only added back in numpy 1.20.
> 
> Changing all the (much older) remaining `where` arguments to keyword-only would be technically be backwards compatible. While I doubt they're used much as a positional parameter out in the wild, it's probably not the risk of breaking things.

I read that as do nothing. I agree that such a change would be backwards incompatible and that one can argue that it is not terribly important. Sometimes it can be helpful to have an issue that others can find if they have the same question, but in this case it it perhaps more of a general problem of kw only args throughout the api.

**Comment by InessaPawson** [comment]
Hi-five on merging your first pull request to NumPy, @janusheide! We hope you stick around! Your choices aren’t limited to programming – you can review pull requests, help us stay on top of new and old  issues, develop educational material, work on our website, add or improve graphic design, create marketing materials, translate website content, write grant proposals, and help with other fundraising initiatives. For more info, check out: https://numpy.org/contribute
Also, consider joining our mailing list. This is a great way to connect with other cool people in our community and be part of important conversations that affect the development of NumPy: https://mail.python.org/mailman/listinfo/numpy-discussion

**Pull request merged**